### PR TITLE
Add minetest.get_objects_in_area

### PR DESCRIPTION
**Pull request by EliasFleckenstein03**
See #10641

This PR is ready for Review.

How to test:
```lua
minetest.register_chatcommand("g", {
	func = function(name)
		local player = minetest.get_player_by_name(name)
		local pos = player:get_pos()
		local objs = minetest.get_objects_in_area(vector.add(pos, 4), vector.add(pos, -4))
		for i, obj in ipairs(objs) do
			local ent = obj:get_luaentity()
			print(i, ent and ent.name or "<player>")
		end 
	end
})
```

### CODE REVIEW

File: src/script/lua_api/l_env.cpp [hunk 1]
```diff
@@ -743,6 +743,31 @@ int ModApiEnvMod::l_get_objects_inside_radius(lua_State *L)
...
 
+// get_objects_in_area(pos, minp, maxp)
+int ModApiEnvMod::l_get_objects_in_area(lua_State *L)
+{
+	GET_ENV_PTR;
+	ScriptApiBase *script = getScriptApiBase(L);
+	
+	v3f minp = read_v3f(L, 1);
+	v3f maxp = read_v3f(L, 2);
+	aabb3f box(minp, maxp);
```
Comment (hecktest) [inline src/script/lua_api/l_env.cpp:754]: This part is missing the `* BS` multiply, which is necessary to convert the position params to node sized units.

**Review by hecktest — CHANGES_REQUESTED** [review]

File: src/script/lua_api/l_env.cpp (hunk 1 above)
Comment (EliasFleckenstein03) [reply src/script/lua_api/l_env.cpp:754]: Thanks, fixed it.

**Review by hecktest — APPROVED** [review]
No more objections, it works.

**Comment by runsy** [comment]
Important for mob mods. To interact with the environment faster.

**Comment by EliasFleckenstein03** [comment]
Have I done something wrong or why is this completely getting ignored by the core devs?

**Comment by Adnan1091** [comment]
Please can someone approve this?

File: doc/lua_api.txt [hunk 2]
```diff
@@ -4864,6 +4864,9 @@ Environment access
 * `minetest.get_objects_inside_radius(pos, radius)`: returns a list of
   ObjectRefs.
     * `radius`: using an euclidean metric
+* `minetest.get_objects_in_area(pos1, pos2)`: returns a list of
+  ObjectRefs.
+     * `pos1` and `pos2` are the min and max positions of the area to search.
```
Comment (SmallJoker) [inline doc/lua_api.txt:4869]:
```suggestion
     * `pos1` and `pos2` are the corner positions of the volume to search.
```

**Review by SmallJoker — APPROVED** [review]
Testing mod. Works.
```Lua
minetest.register_chatcommand("g", {
	func = function(name)
		local player = minetest.get_player_by_name(name)
		local pos = player:get_pos()
		local objs = minetest.get_objects_in_area(vector.add(pos, 4), vector.add(pos, -4))
		for i, obj in ipairs(objs) do
			local ent = obj:get_luaentity()
			print(i, ent and ent.name or "<player>")
		end 
	end
})
```

File: doc/lua_api.txt (hunk 2 above)
Comment (EliasFleckenstein03) [reply doc/lua_api.txt:4869]: `* `minetest.find_nodes_in_area(pos1, pos2, nodenames, [grouped])`
    * `pos1` and `pos2` are the min and max positions of the area to search.`

Comment (EliasFleckenstein03) [reply doc/lua_api.txt:4869]: I just copied the description of `find_nodes_in_area`.

File: src/server/activeobjectmgr.cpp [hunk 3]
```diff
@@ -127,6 +127,22 @@ void ActiveObjectMgr::getObjectsInsideRadius(const v3f &pos, float radius,
 	}
 }
 
+// clang-format on
```
Comment (sfan5) [inline src/server/activeobjectmgr.cpp:130]: This shouldn't be here

File: src/serverenvironment.h [hunk 4]
```diff
@@ -331,6 +331,13 @@ class ServerEnvironment : public Environment
 	{
 		return m_ao_manager.getObjectsInsideRadius(pos, radius, objects, include_obj_cb);
 	}
+	
+	// Find all active objects inside a box
+	void getObjectsInArea(std::vector<ServerActiveObject *> &objects, const aabb3f box,
```
Comment (sfan5) [inline src/serverenvironment.h:336]:
```suggestion
	void getObjectsInArea(std::vector<ServerActiveObject *> &objects, const aabb3f &box,
```

File: src/server/activeobjectmgr.cpp (hunk 3 above)
Comment (EliasFleckenstein03) [reply src/server/activeobjectmgr.cpp]: Changed it.

**Comment by runsy** [comment]
Thanks to the author and reviewers!!! :-)

**Pull request merged**